go-controller/pkg/clustermanager/secondary_network_unit_test.go

```diff
@@ -210,6 +210,86 @@ var _ = ginkgo.Describe("Cluster Controller Manager", func() {
 			})
 		})
 
+		ginkgo.When("Attaching to a localnet network", func() {
+			const subnets = "192.168.200.0/24,fd12:1234::0/64"
+
+			var (
+				fakeClient *util.OVNClusterManagerClientset
+				netInfo    util.NetInfo
+			)
+
+			ginkgo.BeforeEach(func() {
+				fakeClient = &util.OVNClusterManagerClientset{
+					KubeClient:            fake.NewSimpleClientset(&v1.NodeList{Items: nodes()}),
+					NetworkAttchDefClient: fakenadclient.NewSimpleClientset(),
+				}
+			})
+
+			ginkgo.DescribeTable(
+				"the secondary network controller",
+				func(netConf *ovncnitypes.NetConf, featureConfig config.OVNKubernetesFeatureConfig, expectedError error) {
+					var err error
+					netInfo, err = util.NewNetInfo(netConf)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+					app.Action = func(ctx *cli.Context) error {
+						gomega.Expect(initConfig(ctx, featureConfig)).To(gomega.Succeed())
+
+						f, err = factory.NewClusterManagerWatchFactory(fakeClient)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						err = f.Start()
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+						sncm, err := newSecondaryNetworkClusterManager(fakeClient, f, recorder)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+						_, err = sncm.NewNetworkController(netInfo)
+						if expectedError == nil {
+							gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						} else {
+							gomega.Expect(err).To(gomega.MatchError(expectedError))
+						}
+
+						return nil
+					}
+
+					gomega.Expect(app.Run([]string{app.Name})).To(gomega.Succeed())
+				},
+				ginkgo.Entry(
+					"does not manage localnet topologies on IC deployments for networks without subnets",
+					&ovncnitypes.NetConf{NetConf: types.NetConf{Name: "blue"}, Topology: ovntypes.LocalnetTopology},
+					ovnkICConfig(),
```

Comment:
I mean ... this is in a file named "secondary network unit test". 

It's pretty pointless to try to use these features with multi-networks set to false IMHO.

But OK, you both are saying this is convoluted, I'll just shove the config in the readers eyes. 

FWIW, I can argue (for days) that the wrong thing here is having a singleton for the configuration since that impacts **all the other tests**.

Reply:
using the existing `initConfig` config function makes it hard to enable multi network on the before each - i.e. we'd overwrite it on the test's body.

I now carry around both features in all tests.